### ISSUE TITLE
Confusing error message for tuple arguments

### ISSUE THREAD

**Issue by marek-safar** [dotnet/roslyn#20172]
**Version Used**:  2.3.0.61719 (5fbdd5c5)

**Steps to Reproduce**:

```csharp
static class X
{
    static (X, long) Foo<T> ()
    {
    }
}
```

**Expected Behavior**:

Meaningful error message

**Actual Behavior**:

error CS0718: 'X': static types cannot be used as type arguments

**Comment by TyOverby** [comment]
Would this be better if the error message location was at over the "X" and not "Foo"?

**Comment by alrz** [comment]
`C<X, long> Foo()` would also produce error on `Foo`. If we're going to move it to `X`, constraint checking needs to be updated as well — for a case like `unsafe (int*, int) Foo()`.

**Comment by TyOverby** [comment]
@alrz Yeah, I think switching up error location there would be good too.

**Comment by CyrusNajmabadi** [comment]
Note: this is unrelated to tuple arguments.  You get a similar error if you do `static X Foo()` or `static X? Foo()`, etc.  We report the violation on the method, not on the actual static class usage location.
